lib/membrane_h264_plugin/parser.ex

```diff
@@ -54,24 +58,45 @@ defmodule Membrane.H264.Parser do
   @impl true
   def handle_init(opts) do
     state = %{
-      params: opts.sps <> opts.pps,
-      nalu_splitter: NALuSplitter.new(),
+      nalu_splitter: NALuSplitter.new(opts.sps <> opts.pps),
       nalu_parser: NALuParser.new(),
-      au_splitter: AUSplitter.new()
+      au_splitter: AUSplitter.new(),
+      mode: nil,
+      previous_timestamps: {nil, nil}
     }
 
     {:ok, state}
   end
 
   @impl true
-  def handle_caps(:input, _caps, _ctx, state) do
+  def handle_caps(:input, caps, _ctx, state) do
+    mode =
+      case caps do
+        %RemoteStream{type: :bytestream} -> :bytestream
+        %RemoteStream{type: :packetized, content_format: :nalu} -> :nalu_aligned
+        %RemoteStream{type: :packetized, content_format: :au} -> :au_aligned
+      end
+
+    state = %{state | mode: mode}
     {:ok, state}
   end
 
   @impl true
   def handle_process(:input, %Membrane.Buffer{} = buffer, _ctx, state) do
-    payload = state.params <> buffer.payload
-    {nalus_payloads_list, nalu_splitter} = NALuSplitter.split(payload, state.nalu_splitter)
+    {nalus_payloads_list, nalu_splitter} = NALuSplitter.split(buffer.payload, state.nalu_splitter)
```

Comment:
We might consider not calling this in nalu_aligned mode but I don't insist

Reply:
I will leave it that way and we will change it with the introduction of input and output behaviours, as we discussed yesterday